Code/MinimalLib/minilib.cpp

```diff
@@ -642,9 +641,40 @@ std::string JSReaction::get_svg_with_highlights(
   int h = d_defaultHeight;
   return MinimalLib::rxn_to_svg(*d_rxn, w, h, details);
 }
+bool JSReaction::is_valid() const { return true; }
 
-bool JSReaction::is_valid() const {
-  return true;
+std::vector<JSMolList> JSReaction::run_reactants(const JSMolList &reactants,
+                                                unsigned int maxProducts) const {
+  if (!d_rxn) {
+    std::cerr << "Error: d_rxn is null\n";
+    return {};
+  }
+  d_rxn->initReactantMatchers();
+  RDKit::MOL_SPTR_VECT reactant_vec;
+
+  for (const auto &reactant : reactants.mols()) {
+    if (!reactant) {
+      std::cerr << "Error: reactant is null\n";
```

Comment:
We don't do error reporting straight to `std::cerr`. 
I think you need to thrown an exception here, but @ptosco should suggest what he thinks is the best approach for error handling in minilib